maps/src/main/java/com/gluonhq/maps/tile/TileRetriever.java

```diff
@@ -39,6 +41,19 @@
      * @param i the horizontal position of the tile to load
      * @param j the vertical position of the tile to load
      * @return an image representing the tile
+     * @throws IOException If an exception occurs while retrieving the tile
+     */
+    Image loadTile(int zoom, long i, long j) throws IOException;
+
+    /**
+     * Loads a tile from a file cache, at the specified zoom level and
+     * coordinates and returns it as an {@link Image}. Returns null if
+     * such file doesn't exist
+     *
+     * @param zoom the desired zoom level for the tile to load
+     * @param i the horizontal position of the tile to load
+     * @param j the vertical position of the tile to load
+     * @return an image representing the tile
      */
-    Image loadTile(int zoom, long i, long j);
+    Image loadFromCache(int zoom, long i, long j);
```

Comment:
On the interface level, do we need two methods? 
Seems like we can hide the fact that it is coming from cache

Reply:
Problem with one single method (`loadTile`): if the tile is found in the cache, returns the image. If not, we need the async task to download it, so it won't return the image immediately.

So one method is not enough for these two different cases, imo.

Reply:
In case you usually return a Feature<Image> I think.

Reply:
Sure, you can return a `CompletableFuture<Image>`, which is different from returning an `Image`. You can't have both in the same method, can you?

Reply:
No, but I think CompletableFuture as a result is clear definition of delayed operation.

Reply:
Ok, pushed some changes refactoring the Api.